File: back-end/Models/Store.cs
using System.ComponentModel.DataAnnotations;
using System.Text.Json.Serialization;

namespace Efficiency.Models;

public class Store
{
    [Key]
    public int ID { get; set; }
    public string? Name { get; set; }
    public string? City { get; set; }
    public string? State { get; set; }
    public string? Country { get; set; }
    public string? Street { get; set; }
    public string? District { get; set; }
    public string? Complement { get; set; }
    public string? Observations { get; set; }
    public string? ZipCode { get; set; }

    [JsonIgnore]
    public virtual ICollection<User>? Users { get; set; }
    [JsonIgnore]
    public virtual ICollection<Seller>? Sellers { get; set; }
    [JsonIgnore]
    public virtual ICollection<Goal>? Goals { get; set; }
    [JsonIgnore]
    public virtual ICollection<Service>? Services { get; set; }
}

File: front-end/src/app/Models/Store.ts
import { Services } from "src/app/Models/Service";
import { Sellers } from "./Seller";
import { Goals } from "./Goal";
import { Users } from "./User";

export interface Store {
    ID: number;
    Name: string;
    City: string;

    State: string;
    Country: string;
    AddressLine1: string;
    AddressLine2: string;
    ZipCode: string;

    Users: Users;

    Sellers: Sellers;

    Goals: Goals;

    Services: Services;
}

export type Companies = Array<Store>;
